ai/integrations/autogen/tests/test_autogen_toolkit.py

```diff
@@ -3,195 +3,195 @@
     monkeypatch.setenv(USE_SERVERLESS, str(use_serverless))
-    client = get_client()
-    with set_default_client(client), create_function_and_cleanup(client, schema=SCHEMA) as func_obj:
-        toolkit = UCFunctionToolkit(function_names=[func_obj.full_function_name], client=client)
+    with (
+        set_default_client(dbx_client),
+        create_function_and_cleanup(dbx_client, schema=SCHEMA) as func_obj,
+    ):
+        toolkit = UCFunctionToolkit(function_names=[func_obj.full_function_name], client=dbx_client)
         tools = toolkit.tools
         assert len(tools) == 1
         tool = tools[0]
         assert tool.name == func_obj.tool_name
         assert func_obj.comment in tool.description
+
         input_args = {"code": "print(1)"}
-        result = json.loads(tool.fn(**input_args))["value"]
+        result_str = await tool.run_json(input_args)
+        result = json.loads(result_str)["value"]
         assert result == "1\n"
 
-        toolkit = UCFunctionToolkit(
-            function_names=[f.full_name for f in client.list_functions(CATALOG, SCHEMA)],
-            client=client,
+        # Re-check multiple
+        toolkit2 = UCFunctionToolkit(
+            function_names=[f.full_name for f in dbx_client.list_functions(CATALOG, SCHEMA)],
+            client=dbx_client,
         )
-        assert len(toolkit.tools) >= 1
-        assert func_obj.tool_name in [t.name for t in toolkit.tools]
+        assert len(toolkit2.tools) >= 1
+        assert func_obj.tool_name in [t.name for t in toolkit2.tools]
 
 
 @requires_databricks
 @pytest.mark.parametrize("use_serverless", [True, False])
-def test_multiple_toolkits(use_serverless, monkeypatch):
+@pytest.mark.asyncio
+async def test_multiple_toolkits(use_serverless, monkeypatch, dbx_client):
     monkeypatch.setenv(USE_SERVERLESS, str(use_serverless))
-    client = get_client()
-    with set_default_client(client), create_function_and_cleanup(client, schema=SCHEMA) as func_obj:
-        toolkit1 = UCFunctionToolkit(function_names=[func_obj.full_function_name])
+    with (
+        set_default_client(dbx_client),
+        create_function_and_cleanup(dbx_client, schema=SCHEMA) as func_obj,
+    ):
+        toolkit1 = UCFunctionToolkit(
+            function_names=[func_obj.full_function_name], client=dbx_client
+        )
         toolkit2 = UCFunctionToolkit(
-            function_names=[f.full_name for f in client.list_functions(CATALOG, SCHEMA)]
+            function_names=[f.full_name for f in dbx_client.list_functions(CATALOG, SCHEMA)],
+            client=dbx_client,
         )
         tool1 = toolkit1.tools[0]
         tool2 = [t for t in toolkit2.tools if t.name == func_obj.tool_name][0]
+
         input_args = {"code": "print(1)"}
-        result1 = json.loads(tool1.fn(**input_args))["value"]
-        result2 = json.loads(tool2.fn(**input_args))["value"]
+        result1_str = await tool1.run_json(input_args)
```

Comment:
Seems like there's an error here https://github.com/databricks-eng/ai-oss-integration-tests-runner/actions/runs/12821998267/job/35754221891

Reply:
yep - forgot to add that continuation token to those suites. Will update!

Reply:
e2e suites are passing now :D